File: Leetcode/2739. Total Distance Traveled.go
// https://leetcode.com/problems/total-distance-traveled/

func distanceTraveled(mainTank int, additionalTank int) int {
    result := 0
    for mainTank >= 5 && additionalTank > 0 {
        transfered := min(int(mainTank / 5), additionalTank)
        result += 50 * transfered
        mainTank -= 4 * transfered
        additionalTank -= transfered
    }
    return result + mainTank * 10
}

func min(x, y int) int {
    if x < y {
        return x
    }
    return y
}

File: Leetcode/2739. Total Distance Traveled.ts
// https://leetcode.com/problems/total-distance-traveled/

function distanceTraveled(mainTank: number, additionalTank: number): number {
    let result = 0;
    while (mainTank >= 5 && additionalTank > 0) {
        let transfered = Math.min(Math.floor(mainTank / 5), additionalTank);
        result += 50 * transfered;
        mainTank -= 4 * transfered;
        additionalTank -= transfered;
    }
    return result + mainTank * 10;
};